lib/email/receiver.rb

```diff
@@ -601,6 +602,8 @@ def process_destination(destination, user, body, elided, hidden_reason_id)
           raise ReplyUserNotMatchingError, "post_reply_key.user_id => #{post_reply_key.user_id.inspect}, user.id => #{user.id.inspect}"
         end
 
+        enable_email_pm_setting(user)
```

Comment:
I think this will change the setting even when the user replies to a public topic instead of a PM. Probably not a big deal, but worth fixing if it's easy to fix.